const ItemsModel = require('../model/items')
// Create and Save a new user
exports.create = async (req, res) => {
    if (!req.body.name && !req.body.category && !req.body.anime && !req.body.phone) {
        res.status(400).send({ message: "Content can not be empty!" });
    }

    const items = new ItemsModel({
        name: req.body.name,
        category: req.body.category,
        anime: req.body.anime,
    });

    await items.save().then(data => {
        res.send({
            message:"Items created successfully!!",
            items:data
        });
    }).catch(err => {
        res.status(500).send({
            message: err.message || "Some error occurred while creating user"
        });
    });
};
// Retrieve all users from the database.
exports.findAll = async (req, res) => {
    try {
        const items = await ItemsModel.find();
        res.status(200).json(items);
    } catch(error) {
        res.status(404).json({message: error.message});
    }
};
// Find a single User with an id
exports.findOne = async (req, res) => {
    try {
        const items = await ItemsModel.findById(req.params.id);
        res.status(200).json(items);
    } catch(error) {
        res.status(404).json({ message: error.message});
    }
};
// Update a user by the id in the request
exports.update = async (req, res) => {
    if(!req.body) {
        res.status(400).send({
            message: "Data to update can not be empty!"
        });
    }

    const id = req.params.id;

    await ItemsModel.findByIdAndUpdate(id, req.body, { useFindAndModify: false }).then(data => {
        if (!data) {
            res.status(404).send({
                message: `User not found.`
            });
        }else{
            res.send({ message: "User updated successfully." })
        }
    }).catch(err => {
        res.status(500).send({
            message: err.message
        });
    });
};
// Delete a user with the specified id in the request
exports.destroy = async (req, res) => {
    await ItemsModel.findByIdAndRemove(req.params.id).then(data => {
        if (!data) {
            res.status(404).send({
                message: `User not found.`
            });
        } else {
            res.send({
                message: "User deleted successfully!"
            });
        }
    }).catch(err => {
        res.status(500).send({
            message: err.message
        });
    });
};